x-pack/plugins/ingest_pipelines/public/application/sections/pipelines_create/pipelines_create.tsx

```diff
@@ -50,11 +50,11 @@ export const PipelinesCreate: React.FunctionComponent<RouteComponentProps & Prop
       return;
     }
 
-    history.push(BASE_PATH + `?pipeline=${encodeURIComponent(pipeline.name)}`);
+    history.push(URL_GENERATOR[INGEST_PIPELINES_PAGES.LIST](pipeline.name));
```

Comment:
This seems like a bit of an awkward interface because I have to import two things, combine them in the right way, and then infer that they map to a function which I then call. It seems like a more intuitive interface would be to import a function and then call that.

```js
import { getListPath } from '../../services/navigation';

/* ... */

history.push(getListPath(pipeline.name));
```

WDYT?

EDIT: I just got to the end of the PR and saw the URL generator, which implements an interface similar to what I'm suggesting. Is it possible to use the generator internally?

```js
import { urlGenerator } from '../../services/navigation';

/* ... */

history.push(getListPath(pipeline.name));
history.push(urlGenerator.createUrl({
  page: urlGenerator.LIST,
  pipelineId: pipeline.name,
}));
```

Reply:
I agree, the import is too complicated, I changed it to a function export for internal use. I'm not sure `UrlGenerator` should be used internally, since we don't need an absolute path and `createUrl` is async.